src/test/java/org/jenkinsci/plugins/workflow/libs/LibraryMemoryTest.java

```diff
@@ -82,7 +92,44 @@ public static void register(Object o) {
             clearInvocationCaches.invoke(metaClass);
         }
         for (WeakReference<ClassLoader> loaderRef : LOADERS) {
-            MemoryAssert.assertGC(loaderRef);
+            MemoryAssert.assertGC(loaderRef, false);
+        }
+    }
+
+    @Test public void loaderReleasedWithGrab() throws Exception {
```

Comment:
Without https://github.com/jenkinsci/workflow-cps-plugin/pull/1086, this test either passes very slowly, or fails, depending on whether or not the `SoftReference` is cleared prior to an `OutOfMemoryError`. With https://github.com/jenkinsci/jenkins-test-harness/pull/1029 the failure becomes consistent.